base_serializer.go

```diff
@@ -24,48 +30,155 @@ type baseSerializer struct {
 	hasContainerAnnotation bool
 }
 
-func (baseSerializer *baseSerializer) stepIn(ionValue interface{}) {
-	panic("implement me")
+func (baseSerializer *baseSerializer) stepIn(ionValue hashValue) error {
+	err := baseSerializer.handleFieldName(ionValue)
+	if err != nil {
+		return err
+	}
+
+	err = baseSerializer.handleAnnotationsBegin(ionValue, true)
+	if err != nil {
+		return err
+	}
+
+	baseSerializer.beginMarker()
+
+	tq := tq(ionValue)
+	if ionValue.isNull() {
+		tq = tq | 0x0F
+	}
+
+	baseSerializer.update([]byte{tq})
+
+	return nil
 }
 
-func (baseSerializer *baseSerializer) digest() []byte {
-	panic("implement me")
+func (baseSerializer *baseSerializer) Sum(b []byte) []byte {
+	return baseSerializer.hashFunction.Sum(b)
 }
 
-func (baseSerializer *baseSerializer) handleFieldName(ionValue interface{}) {
-	panic("implement me")
+func (baseSerializer *baseSerializer) handleFieldName(ionValue hashValue) error {
+	if baseSerializer.depth > 0 && ionValue.isInStruct() {
+		// SymbolTokens are not available right now.
```

Comment:
Can we denote future work with a TODO? Please also check for this in other places in this commit that makes sense.

e.g., TODO SymbolTokens are not available right now.

Reply:
Done.